actionview/lib/action_view/renderer/partial_renderer.rb

```diff
@@ -307,7 +307,8 @@ def render(context, options, block)
       if @collection
         render_collection
       else
-        instrument(:partial) do
+        instrument(:partial) do |payload|
+          @view.instance_variable_set(:@log_payload_for_partial_render, payload)
           render_partial
```

Comment:
Okay, you we're right before. Moving the `instrument` to `render_partial` is better such that we only touch the `@view` from in there and have more consistency with `render_collection`.